### DISCUSSION TITLE
Is there a rule that requires "window." when accessing global?

### DISCUSSION THREAD

**Discussion by keith0305** [eslint/eslint#16912] [Q&A]
Wrong code under this rule:
```js
const href = location.href;
```

Correct code under this rule:
```js
const href = window.location.href;
```
Is there such a rule already?
I can't seem to find such rule in the doc, but I might be missing it (the search isn't exactly very friendly).

**Comment by Rec0iL99** [comment]
Hi @keith0305, you can use the [no-restricted-globals](https://eslint.org/docs/latest/rules/no-restricted-globals) to restrict certain global variables. Here's how you do it:

```js
{
    rules: {
        no-restricted-globals: ["error", "location"]
    }
}
```

You can add more global variables you want to restrict to the list as a string.

Here's a [playground demo](https://eslint.org/play/#eyJ0ZXh0IjoiLyogZXNsaW50IG5vLXJlc3RyaWN0ZWQtZ2xvYmFsczogW1wiZXJyb3JcIiwgXCJsb2NhdGlvblwiXSAqL1xuXG4vLyBpbmNvcnJlY3QgY29kZVxuY29uc3QgaHJlZkluY29ycmVjdCA9IGxvY2F0aW9uLmhyZWY7IFxuXG4vLyBjb3JyZWN0IGNvZGVcbmNvbnN0IGhyZWZDb3JyZWN0ID0gd2luZG93LmxvY2F0aW9uLmhyZWY7XG5cblxuIiwib3B0aW9ucyI6eyJwYXJzZXJPcHRpb25zIjp7ImVjbWFWZXJzaW9uIjo2LCJzb3VyY2VUeXBlIjoic2NyaXB0IiwiZWNtYUZlYXR1cmVzIjp7fX0sInJ1bGVzIjp7fSwiZW52Ijp7ImJyb3dzZXIiOnRydWV9fX0=) working for your specific case.

Thanks for using ESLint

**Reply by keith0305** [reply]
Was wondering on the practicality of this. There are hundreds of properties on the window object, and browser vendors might ship new properties in the future.

I was thinking is there a way to restrict to allow only certain globals instead?

**Reply by Rec0iL99** [reply]
You can get creative by cloning the existing rule and creating a custom rule. 

Get all the keys of `window` using `Object.keys`.

```js
Object.keys(window)
// expected output
['window', 'self', 'document', 'name', 'location', 'customElements', ....]
```

Now you can plug these into the `no-restricted-globals` rule config array.

I know this looks silly but ESLint does not support a rule that requires global as far as I know.

**Reply by aladdin-add** [reply]
eslint does not assume your env, so `window` is not a global by default. 


are you using `env: {browser: true}`? I recommended you can set `browser: false`, and then add your own globals `window`.

**Reply by eMerzh** [reply]
@aladdin-add seems a bbity strange to me... it's not that the env is not a browser, but we wat to make it clear where the var comes from.

if i do that seems that all my types are flagged as issues ... 

i recently had an issue where i did 
```js
status == value
```

status came from window and i didn't noticed...
